PrismaDB-QueryAST/DML/Expressions/ScalarFunction.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+
+namespace PrismaDB.QueryAST.DML
+{
+    public class ScalarFunction : Expression
+    {
+        public string FunctionName;
+        public List<Expression> Parameters;
+
+        public ScalarFunction(string functionName)
+        {
+            setValue(functionName, new List<Expression>());
+        }
+
+        public ScalarFunction(string functionName, List<Expression> parameters)
+        {
+            setValue(functionName, parameters);
+        }
+
+        public override Expression Clone()
+        {
+            var clone = new ScalarFunction(FunctionName, new List<Expression>(Parameters));
+
+            return clone;
+        }
+
+        public override bool Equals(object other)
+        {
+            var otherF = other as ScalarFunction;
+            if (otherF == null) return false;
+
+            if (Parameters.Count == 0 && otherF.Parameters.Count == 0)
+                return FunctionName.Equals(otherF.FunctionName);
+            else
+                return FunctionName.Equals(otherF.FunctionName)
+                    && ((Parameters.Count == otherF.Parameters.Count)
+                    && Enumerable.Zip(Parameters, otherF.Parameters, (x, y)
+                    => x.Equals(y)).Aggregate((x, y) => x && y));
+        }
+
+        public override object Eval(DataRow r)
+        {
+            return r[ToString()];
+        }
+
+        public override List<ColumnRef> GetColumns()
+        {
+            return new List<ColumnRef>();
+        }
+
+        public override int GetHashCode()
+        {
+            return unchecked(FunctionName.GetHashCode() *
+                Parameters.Aggregate((a, b) =>
+                new IntConstant(a.GetHashCode() * b.GetHashCode())).GetHashCode());
```

Comment:
no need to do `new IntConstant` here. This is an override of `object.GetHashCode()`, without which you can't use an object of the class in a Dictionary as a key.